src/node.js

```diff
@@ -539,7 +541,12 @@
         rejectionHandled(promise);
       else
         NativeModule.require('assert').fail('unexpected PromiseRejectEvent');
-    });
+    }, function onUnhandledReject() {
+      if (!process.emit('unhandledRejection', reason, promise)) {
```

Comment:
@Fishrock123 note that `reason` is not in scope here. Once you fix the `ToLocalChecked()` issue, you will run into this.